src/db/car/plain.rs

```diff
@@ -162,21 +166,42 @@ impl<ReaderT: super::RandomAccessFileReader> PlainCar<ReaderT> {
                     version,
                     header_v1,
                     header_v2,
+                    metadata: OnceLock::new(),
                 })
             }
         }
     }
 
-    pub fn roots(&self) -> &NonEmpty<Cid> {
-        &self.header_v1.roots
+    pub fn metadata(&self) -> &Option<FilecoinSnapshotMetadata> {
+        self.metadata.get_or_init(|| {
+            if self.header_v1.roots.len() == super::V2_SNAPSHOT_ROOT_COUNT {
+                let maybe_metadata_cid = self.header_v1.roots.first();
+                if let Ok(Some(metadata)) =
+                    self.get_cbor::<FilecoinSnapshotMetadata>(maybe_metadata_cid)
+                {
+                    return Some(metadata);
+                }
+            }
+            None
+        })
+    }
+
+    pub fn head_tipset_key(&self) -> &NonEmpty<Cid> {
+        // head tipset key is stored in v2 snapshot metadata
+        // See <https://github.com/filecoin-project/FIPs/blob/master/FRCs/frc-0108.md#v2-specification>
```

Comment:
It should be more robust to paste links as permalinks in case something changes.

Reply:
Fixed.